src/dom/set.js

```diff
@@ -8,6 +8,11 @@ function set (subject, property, value) {
 	if (property in setProps) {
 		setProps[property](subject, value);
 	}
+        else if (property === "dataset") {
+		for (const key of Object.keys(value)) {
+			subject.dataset[key]=value[key]
+		}
+        }
```

Comment:
Nit: Please follow the code style of the rest of the codebase (smart tabs for indentation, space around operators, etc.). I need to add an ESLint file…